exporting/send_data.c

```diff
@@ -53,7 +57,39 @@ void simple_connector_receive_response(int *sock, struct instance *instance)
         buffer_need_bytes(response, 4096);
 
         ssize_t r;
+#ifdef ENABLE_HTTPS
+        if (connector_specific_config->conn && connector_specific_config->flags == NETDATA_SSL_HANDSHAKE_COMPLETE) {
+            r = (ssize_t)SSL_read(connector_specific_config->conn,
+                                  &response->buffer[response->len],
+                             (int) (response->size - response->len));
+
+            if (likely(r > 0)) {
+                // we received some data
+                response->len += r;
+                stats->received_bytes += r;
+                stats->receptions++;
+                continue;
+            } else {
+                int sslerrno = SSL_get_error(connector_specific_config->conn, (int) r);
+                u_long sslerr = ERR_get_error();
+                char buf[256];
+                switch (sslerrno) {
+                    case SSL_ERROR_WANT_READ:
+                    case SSL_ERROR_WANT_WRITE:
+                        goto endloop;
```

Comment:
Isn't it better to use `break`?

Reply:
Break here will only break the switch, but we have a while loop outside.

Reply:
Ah, sorry, I didn't notice the switch.